### ISSUE TITLE
Wrong Outputs from pytorch3d.transforms.so3_log_map

### ISSUE THREAD

**Issue by akarshkumar0101** [facebookresearch/pytorch3d#413]
Hi! Thanks for the library and the support you give! 

For certain inputs, the function `pytorch3d.transforms.so3_log_map` does not give the right answer.

For the following rotation matrix (flip x-axis and swap y and z axis)
-1, 0, 0
 0, 0, 1
 0, 1, 0
it returns an axis-angle representation of [0, 0, 0] (same as identity rotation matrix).
It should return [0, -2.2214, -2.2214] as this is the correct axis-angle representation of that matrix.

Code:
```
import torch
import pytorch3d.transforms

R = torch.tensor([[[-1.,  0.,  0.],
                   [ 0.,  0.,  1.],
                   [ 0.,  1.,  0.]]])
print(pytorch3d.transforms.so3_log_map(R))

axang = torch.tensor([[-0.0000, -2.2214, -2.2214]])
pytorch3d.transforms.so3_exponential_map(axang)
```

Output:

```
tensor([[-0., -0., -0.]])
tensor([[[-1.0000e+00,  4.1411e-05, -4.1411e-05],
         [-4.1411e-05,  0.0000e+00,  1.0000e+00],
         [ 4.1411e-05,  1.0000e+00,  0.0000e+00]]])
```

Thanks!

**Comment by davnov134** [comment]
Hi @akarshkumar0101 ,

thanks for your question.

Such outputs of `so3_log_map` are (unfortunately) expected since the conversion has a singularity in cases the sine of the rotation angle is close to zero. For your case:
```
print(pytorch3d.transforms.so3_rotation_angle(R).sin())  # tensor([-8.7423e-08])
```
You can try to adjust the `eps` argument of the `so3_log_map` in case you are too close to this singular case.

A similar issue has been raised here: #262
